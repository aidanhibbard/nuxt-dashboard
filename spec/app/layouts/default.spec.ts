import { describe, it, expect } from 'vitest'
import { mount } from '@vue/test-utils'
import DefaultLayout from '@/layouts/default.vue'

describe('layouts/default.vue', () => {
  it('renders header, sidebar and slots content', () => {
    const wrapper = mount(DefaultLayout, {
      slots: {
        default: '<div id="page-content">Hello</div>',
      },
    })

    // Sidebar component placeholder should exist via AppSidebar mount
    expect(wrapper.findComponent({ name: 'AppSidebar' }).exists()).toBe(true)
    // Header component placeholder should exist via AppHeader mount
    expect(wrapper.findComponent({ name: 'AppHeader' }).exists()).toBe(true)
    // Slot content rendered
    expect(wrapper.find('#page-content').text()).toBe('Hello')
  })

  it('matches snapshot', () => {
    const wrapper = mount(DefaultLayout, { slots: { default: '<div />' } })
    expect(wrapper.html()).toMatchSnapshot()
  })
})
