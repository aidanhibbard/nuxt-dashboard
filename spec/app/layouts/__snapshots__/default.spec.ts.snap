// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`layouts/default.vue > matches snapshot 1`] = `
"<div>
  <aside collapsible="icon">
    <header>
      <ul>
        <li>
          <div>
            <div as-child=""><button size="lg" class="data-[state=open]:bg-sidebar-accent data-[state=open]:text-sidebar-accent-foreground">
                <div class="flex aspect-square size-8 items-center justify-center rounded-lg bg-sidebar-primary text-sidebar-primary-foreground"><svg xmlns="http://www.w3.org/2000/svg" width="24" height="24" viewBox="0 0 24 24" fill="none" stroke="currentColor" stroke-width="2" stroke-linecap="round" stroke-linejoin="round" class="lucide size-4 lucide-gallery-vertical-end-icon lucide-gallery-vertical-end size-4">
                    <path d="M7 2h10"></path>
                    <path d="M5 6h14"></path>
                    <rect width="18" height="12" x="3" y="10" rx="2"></rect>
                  </svg></div>
                <div class="grid flex-1 text-left text-sm leading-tight"><span class="truncate font-medium">Acme Inc</span><span class="truncate text-xs">Enterprise</span></div><svg xmlns="http://www.w3.org/2000/svg" width="24" height="24" viewBox="0 0 24 24" fill="none" stroke="currentColor" stroke-width="2" stroke-linecap="round" stroke-linejoin="round" class="lucide ml-auto lucide-chevrons-up-down-icon lucide-chevrons-up-down ml-auto">
                  <path d="m7 15 5 5 5-5"></path>
                  <path d="m7 9 5-5 5 5"></path>
                </svg>
              </button></div>
            <div data-side="right" data-align="start" class="w-[--reka-dropdown-menu-trigger-width] min-w-56 rounded-lg" side-offset="4">
              <div class="text-xs text-muted-foreground"> Teams </div>
              <div class="dropdown-menu-item gap-2 p-2">
                <div class="flex size-6 items-center justify-center rounded-sm border"><svg xmlns="http://www.w3.org/2000/svg" width="24" height="24" viewBox="0 0 24 24" fill="none" stroke="currentColor" stroke-width="2" stroke-linecap="round" stroke-linejoin="round" class="lucide size-3.5 shrink-0 lucide-gallery-vertical-end-icon lucide-gallery-vertical-end size-3.5 shrink-0">
                    <path d="M7 2h10"></path>
                    <path d="M5 6h14"></path>
                    <rect width="18" height="12" x="3" y="10" rx="2"></rect>
                  </svg></div> Acme Inc <kbd>⌘1</kbd>
              </div>
              <div class="dropdown-menu-item gap-2 p-2">
                <div class="flex size-6 items-center justify-center rounded-sm border"><svg xmlns="http://www.w3.org/2000/svg" width="24" height="24" viewBox="0 0 24 24" fill="none" stroke="currentColor" stroke-width="2" stroke-linecap="round" stroke-linejoin="round" class="lucide size-3.5 shrink-0 lucide-audio-waveform-icon lucide-audio-waveform size-3.5 shrink-0">
                    <path d="M2 13a2 2 0 0 0 2-2V7a2 2 0 0 1 4 0v13a2 2 0 0 0 4 0V4a2 2 0 0 1 4 0v13a2 2 0 0 0 4 0v-4a2 2 0 0 1 2-2"></path>
                  </svg></div> Acme Corp. <kbd>⌘2</kbd>
              </div>
              <div class="dropdown-menu-item gap-2 p-2">
                <div class="flex size-6 items-center justify-center rounded-sm border"><svg xmlns="http://www.w3.org/2000/svg" width="24" height="24" viewBox="0 0 24 24" fill="none" stroke="currentColor" stroke-width="2" stroke-linecap="round" stroke-linejoin="round" class="lucide size-3.5 shrink-0 lucide-command-icon lucide-command size-3.5 shrink-0">
                    <path d="M15 6v12a3 3 0 1 0 3-3H6a3 3 0 1 0 3 3V6a3 3 0 1 0-3 3h12a3 3 0 1 0-3-3"></path>
                  </svg></div> Evil Corp. <kbd>⌘3</kbd>
              </div>
              <hr>
              <div class="dropdown-menu-item gap-2 p-2">
                <div class="flex size-6 items-center justify-center rounded-md border bg-transparent"><svg xmlns="http://www.w3.org/2000/svg" width="24" height="24" viewBox="0 0 24 24" fill="none" stroke="currentColor" stroke-width="2" stroke-linecap="round" stroke-linejoin="round" class="lucide size-4 lucide-plus-icon lucide-plus size-4">
                    <path d="M5 12h14"></path>
                    <path d="M12 5v14"></path>
                  </svg></div>
                <div class="font-medium text-muted-foreground"> Add team </div>
              </div>
            </div>
          </div>
        </li>
      </ul>
    </header>
    <section>
      <div>
        <div>Platform</div>
        <ul>
          <div as-child="" default-open="true" class="group/collapsible">
            <li><button as-child="" tooltip="Dashboard"><a href="/"><svg xmlns="http://www.w3.org/2000/svg" width="24" height="24" viewBox="0 0 24 24" fill="none" stroke="currentColor" stroke-width="2" stroke-linecap="round" stroke-linejoin="round" class="lucide lucide-layout-dashboard-icon lucide-layout-dashboard">
                    <rect width="7" height="9" x="3" y="3" rx="1"></rect>
                    <rect width="7" height="5" x="14" y="3" rx="1"></rect>
                    <rect width="7" height="9" x="14" y="12" rx="1"></rect>
                    <rect width="7" height="5" x="3" y="16" rx="1"></rect>
                  </svg><span>Dashboard</span></a></button></li>
          </div>
          <div as-child="" class="group/collapsible">
            <li><button as-child="" tooltip="Users"><a href="/users"><svg xmlns="http://www.w3.org/2000/svg" width="24" height="24" viewBox="0 0 24 24" fill="none" stroke="currentColor" stroke-width="2" stroke-linecap="round" stroke-linejoin="round" class="lucide lucide-users-icon lucide-users">
                    <path d="M16 21v-2a4 4 0 0 0-4-4H6a4 4 0 0 0-4 4v2"></path>
                    <path d="M16 3.128a4 4 0 0 1 0 7.744"></path>
                    <path d="M22 21v-2a4 4 0 0 0-3-3.87"></path>
                    <circle cx="9" cy="7" r="4"></circle>
                  </svg><span>Users</span></a></button></li>
          </div>
          <div as-child="" class="group/collapsible">
            <li><button as-child="" tooltip="Reports"><a href="/reports"><svg xmlns="http://www.w3.org/2000/svg" width="24" height="24" viewBox="0 0 24 24" fill="none" stroke="currentColor" stroke-width="2" stroke-linecap="round" stroke-linejoin="round" class="lucide lucide-file-text-icon lucide-file-text">
                    <path d="M15 2H6a2 2 0 0 0-2 2v16a2 2 0 0 0 2 2h12a2 2 0 0 0 2-2V7Z"></path>
                    <path d="M14 2v4a2 2 0 0 0 2 2h4"></path>
                    <path d="M10 9H8"></path>
                    <path d="M16 13H8"></path>
                    <path d="M16 17H8"></path>
                  </svg><span>Reports</span></a></button></li>
          </div>
          <div as-child="" class="group/collapsible">
            <li><button as-child="" tooltip="Settings"><a href="/settings"><svg xmlns="http://www.w3.org/2000/svg" width="24" height="24" viewBox="0 0 24 24" fill="none" stroke="currentColor" stroke-width="2" stroke-linecap="round" stroke-linejoin="round" class="lucide lucide-settings-icon lucide-settings">
                    <path d="M9.671 4.136a2.34 2.34 0 0 1 4.659 0 2.34 2.34 0 0 0 3.319 1.915 2.34 2.34 0 0 1 2.33 4.033 2.34 2.34 0 0 0 0 3.831 2.34 2.34 0 0 1-2.33 4.033 2.34 2.34 0 0 0-3.319 1.915 2.34 2.34 0 0 1-4.659 0 2.34 2.34 0 0 0-3.32-1.915 2.34 2.34 0 0 1-2.33-4.033 2.34 2.34 0 0 0 0-3.831A2.34 2.34 0 0 1 6.35 6.051a2.34 2.34 0 0 0 3.319-1.915"></path>
                    <circle cx="12" cy="12" r="3"></circle>
                  </svg><span>Settings</span></a></button></li>
          </div>
        </ul>
      </div>
    </section>
    <footer>
      <ul>
        <li>
          <div>
            <div as-child=""><button size="lg" class="data-[state=open]:bg-sidebar-accent data-[state=open]:text-sidebar-accent-foreground">
                <div class="avatar h-8 w-8 rounded-lg"><img src="/avatars/shadcn.jpg" alt="shadcn">
                  <div class="fallback rounded-lg"> CN </div>
                </div>
                <div class="grid flex-1 text-left text-sm leading-tight"><span class="truncate font-medium">shadcn</span><span class="truncate text-xs">m@example.com</span></div><svg xmlns="http://www.w3.org/2000/svg" width="24" height="24" viewBox="0 0 24 24" fill="none" stroke="currentColor" stroke-width="2" stroke-linecap="round" stroke-linejoin="round" class="lucide ml-auto size-4 lucide-chevrons-up-down-icon lucide-chevrons-up-down ml-auto size-4">
                  <path d="m7 15 5 5 5-5"></path>
                  <path d="m7 9 5-5 5 5"></path>
                </svg>
              </button></div>
            <div data-side="right" data-align="end" class="w-[--reka-dropdown-menu-trigger-width] min-w-56 rounded-lg" side-offset="4">
              <div class="p-0 font-normal">
                <div class="flex items-center gap-2 px-1 py-1.5 text-left text-sm">
                  <div class="avatar h-8 w-8 rounded-lg"><img src="/avatars/shadcn.jpg" alt="shadcn">
                    <div class="fallback rounded-lg"> CN </div>
                  </div>
                  <div class="grid flex-1 text-left text-sm leading-tight"><span class="truncate font-semibold">shadcn</span><span class="truncate text-xs">m@example.com</span></div>
                </div>
              </div>
              <hr>
              <div>
                <div class="dropdown-menu-item"><svg xmlns="http://www.w3.org/2000/svg" width="24" height="24" viewBox="0 0 24 24" fill="none" stroke="currentColor" stroke-width="2" stroke-linecap="round" stroke-linejoin="round" class="lucide lucide-sparkles-icon lucide-sparkles">
                    <path d="M11.017 2.814a1 1 0 0 1 1.966 0l1.051 5.558a2 2 0 0 0 1.594 1.594l5.558 1.051a1 1 0 0 1 0 1.966l-5.558 1.051a2 2 0 0 0-1.594 1.594l-1.051 5.558a1 1 0 0 1-1.966 0l-1.051-5.558a2 2 0 0 0-1.594-1.594l-5.558-1.051a1 1 0 0 1 0-1.966l5.558-1.051a2 2 0 0 0 1.594-1.594z"></path>
                    <path d="M20 2v4"></path>
                    <path d="M22 4h-4"></path>
                    <circle cx="4" cy="20" r="2"></circle>
                  </svg> Upgrade to Pro </div>
              </div>
              <hr>
              <div>
                <div class="dropdown-menu-item"><svg xmlns="http://www.w3.org/2000/svg" width="24" height="24" viewBox="0 0 24 24" fill="none" stroke="currentColor" stroke-width="2" stroke-linecap="round" stroke-linejoin="round" class="lucide lucide-badge-check-icon lucide-badge-check">
                    <path d="M3.85 8.62a4 4 0 0 1 4.78-4.77 4 4 0 0 1 6.74 0 4 4 0 0 1 4.78 4.78 4 4 0 0 1 0 6.74 4 4 0 0 1-4.77 4.78 4 4 0 0 1-6.75 0 4 4 0 0 1-4.78-4.77 4 4 0 0 1 0-6.76Z"></path>
                    <path d="m9 12 2 2 4-4"></path>
                  </svg> Account </div>
                <div class="dropdown-menu-item"><svg xmlns="http://www.w3.org/2000/svg" width="24" height="24" viewBox="0 0 24 24" fill="none" stroke="currentColor" stroke-width="2" stroke-linecap="round" stroke-linejoin="round" class="lucide lucide-credit-card-icon lucide-credit-card">
                    <rect width="20" height="14" x="2" y="5" rx="2"></rect>
                    <line x1="2" x2="22" y1="10" y2="10"></line>
                  </svg> Billing </div>
                <div class="dropdown-menu-item"><svg xmlns="http://www.w3.org/2000/svg" width="24" height="24" viewBox="0 0 24 24" fill="none" stroke="currentColor" stroke-width="2" stroke-linecap="round" stroke-linejoin="round" class="lucide lucide-bell-icon lucide-bell">
                    <path d="M10.268 21a2 2 0 0 0 3.464 0"></path>
                    <path d="M3.262 15.326A1 1 0 0 0 4 17h16a1 1 0 0 0 .74-1.673C19.41 13.956 18 12.499 18 8A6 6 0 0 0 6 8c0 4.499-1.411 5.956-2.738 7.326"></path>
                  </svg> Notifications </div>
              </div>
              <hr>
              <div class="dropdown-menu-item"><svg xmlns="http://www.w3.org/2000/svg" width="24" height="24" viewBox="0 0 24 24" fill="none" stroke="currentColor" stroke-width="2" stroke-linecap="round" stroke-linejoin="round" class="lucide lucide-log-out-icon lucide-log-out">
                  <path d="m16 17 5-5-5-5"></path>
                  <path d="M21 12H9"></path>
                  <path d="M9 21H5a2 2 0 0 1-2-2V5a2 2 0 0 1 2-2h4"></path>
                </svg> Log out </div>
            </div>
          </div>
        </li>
      </ul>
    </footer>
    <div class="rail"></div>
  </aside>
  <div>
    <header class="flex h-16 shrink-0 items-center gap-2 transition-[width,height] ease-linear group-has-data-[collapsible=icon]/sidebar-wrapper:h-12">
      <div class="flex items-center gap-2 px-4"><button class="sidebar-trigger -ml-1"></button>
        <hr orientation="vertical" class="mr-2 data-[orientation=vertical]:h-4">
        <nav class="breadcrumb">
          <ul>
            <li class="hidden md:block"><a href="#"> Building Your Application </a></li><span class="hidden md:block">|</span>
            <li><span class="page">Data Fetching</span></li>
          </ul>
        </nav>
      </div>
    </header>
    <div class="flex flex-1 flex-col gap-4 p-4 pt-0">
      <div></div>
    </div>
  </div>
  <div data-v-b0807160="" class="fixed top-4 right-4 z-50 space-y-2">
    <transition-group-stub data-v-b0807160="" name="toast" tag="div" appear="false" persisted="false" css="true" class="space-y-2"></transition-group-stub>
  </div>
</div>"
`;
